node/as/thrift.js

```diff
@@ -40,7 +40,7 @@ function TChannelAsThrift(opts) {
     assert(opts && opts.spec, 'TChannelAsThrift expected spec');
     self.spec = opts.spec;
 
-    self.logger = null;
+    self.logger = opts.logger || null;
```

Comment:
Pulling the logger off of things in `.send` and `.register` is a super weird pattern imo